hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +272,203 @@ default Table newHmsTable(String hmsTableOwner) {
 
     return newTable;
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  default void commitWithHiveLock(
```

Comment:
This is a common code for committing `Iceberg-Table` and `Iceberg-View `on `Hive`.